File: src/main/scala/com/projextxy/core/blocks/item/ItemBlockMetaHandler.java
package com.projextxy.core.blocks.item;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.block.Block;
import net.minecraft.item.ItemBlock;
import net.minecraft.item.ItemStack;

public class ItemBlockMetaHandler extends ItemBlock {
    public ItemBlockMetaHandler(Block block) {
        super(block);
        setHasSubtypes(true);
        setMaxDamage(0);
    }

    @Override
    public int getMetadata(int i) {
        return i;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public String getUnlocalizedName(ItemStack stack) {
        return super.getUnlocalizedName(stack) + "|" + stack.getItemDamage();
    }

}

File: src/main/scala/com/projextxy/core/blocks/item/ItemBlockMetaHandler.scala
package com.projextxy.core.blocks.item

import net.minecraft.block.Block
import net.minecraft.item.{ItemStack, ItemBlock}

class ItemBlockMetaHandler(block: Block) extends ItemBlock(block) {
  setHasSubtypes(true)
  setMaxDamage(0)


  override def getMetadata(meta : Int): Int = meta

  override def getUnlocalizedName(stack: ItemStack): String = s"${super.getUnlocalizedName(stack)}|${stack.getItemDamage}"
}
